File: test-case/floating/array/Numeric.scala
import soda.scala.unittest._

import scala.collection.mutable
import scala.reflect.runtime.universe.typeOf

class Solution {}
object Solution {
  def multiply(a: Array[Double], b: Array[Double]): List[Double] = {
    val res = Array.fill[Double](a.size)(0.0)
    for (i <- res.indices) {
      res(i) = a(i) * b(i)
    }
    res.toList
  }
}

class Numeric extends (String => String) {
  override def apply(text: String): String = {
    val work = TestWork.forObject(typeOf[Solution], "multiply")
    // val work = TestWork.forStruct(...)
    // work.setValidator((R, R) => Boolean)
    // work.compareSerial = true
    work.run(text)
  }
}

object Numeric extends App {
  println(new Numeric()(Utils.fromStdin()))
}

File: test-case/floating/array/Numeric.cs
using Soda.Unittest;

public class Solution {
    public IList<double> Multiply(double[] a, double[] b) {
        var res = new double[a.Length];
        for (int i = 0; i < res.Length; ++i) {
            res[i] = a[i] * b[i];
        }
        return res.ToList();
    }
}

public class Numeric
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().Multiply));
        // work.SetValidator((e, r) => ...);
        // work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}
